sources/appendix-value-extraction.asciidoc

```diff
@@ -0,0 +1,385 @@
+@Retention(RUNTIME)
+public @interface ConstraintsApplyTo {
+    ValidationTarget value();
+}
+----
+
+[source,java]
+----
+public enum ValidationTarget {
+    ANNOTATED_ELEMENT,
+    WRAPPED_VALUE;
+}
+----
+
+Value extractor definitions can be marked with `ConstraintsApplyTo` so that constraints are automatically applied to the wrapped value if a constraint is found for an element handled by that extractor:
+
+[source,java]
+----
+@ConstraintsApplyTo(WRAPPED_VALUE)
+class PropertyExtractor implements ValueExtractor<Property<@ExtractedValue>> {
+
+	@Override
+	public void extractValues(Property[] originalValue, ValueReceiver receiver) {
+		// ...
+	}
+}
+----
+
+If this extractor was the single most-specific extractor for `StringProperty`, the `@Email` constraint above would automatically be applied to the wrapped string value.
+
+In rare cases it may happen that a constraint should be applied to the wrapped value although an extractor exists. In this case the `ConstraintsApplyTo` annotation can be given on the element itself to prevent the unwrapping:
+
+[source,java]
+----
+@ConstraintsApplyTo(ANNOTATED_ELEMENT)
+@NotNull StringProperty emailProperty;
+----
+
+[NOTE]
+.Limitations
+====
+When looking for an extractor in this case, only the extracted types can be considered (e.g. `Map`), but not the extracted type parameters (e.g. Map's `K` or `V` parameters). If there are multiple extractors for the same most specific type (e.g. the `Map` key and value extractors), no extractor can be chosen and an exception will be raised. I.e. automatic unwrapping is only supported for cases where the super-type in question has a single type-parameter.
+====
+
+==== Examples
+
+* Applying a constraint to the value wrapped by a container type:
+
+[source,java]
+----
+Property<@Min(1) Integer> value;
+----
+
+Note that `@Valid` is not required; the `@Min` constraint will be validated when the `value` property is subject to validation.
+
+* Applying a constraint to each value in a collection type:
+
+[source,java]
+----
+List<@NotNull @Email String> emails;
+----
+
+* Cascaded validation of the values in a collection type:
+
+[source,java]
+----
+List<@Valid Order> orders;
+----
+
+* The legacy style for cascaded validation is supported as well:
+
+[source,java]
+----
+@Valid List<Order> orders;
+----
+
+This would also validate any constraints on a custom list type (e.g. `MyList#getId()`). TODO: we never clarified that in 1.1. Should it be made explicite?
+
+* Map validation with type-level constraints and cascading:
+
+[source,java]
+----
+@Valid
+Map<@RegExp(...) String, @RetailOrder Order> orders;
+----
+
+This would validate the map's key's against `@RegExp`, the map's values against `@RetailOrder` and apply cascaded validation of the map (values).
+
+* When selecting extractors, type parameters must be throroughly traced in the hierarchy. Consider this case where the order of the type parameters of `Map` is swapped in a sub-type:
```

Comment:
you used to write type parameters with an hyphen. I think we should be consistent. Personally, I vote for removing the hyphen. See the glossary here: http://beanvalidation.org/glossary/